warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/TableConfigHelperFactoryTest.java

```diff
@@ -6,29 +6,48 @@
 import datawave.util.TableName;
 import org.apache.accumulo.core.client.Connector;
 import org.apache.accumulo.core.client.admin.TableOperations;
-import org.apache.accumulo.core.client.mock.MockInstance;
-import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.accumulo.minicluster.MiniAccumuloCluster;
+import org.apache.accumulo.minicluster.MiniAccumuloConfig;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.log4j.Logger;
+import org.junit.After;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TestName;
 
 import static org.hamcrest.core.Is.is;
 import static org.hamcrest.core.IsNull.nullValue;
 import static org.junit.Assert.assertThat;
 
+import java.io.File;
+
+/**
+ * Each Test starts and stops a mini accumulo cluster. Files are stored in
+ * warehouse/ingest-core/target/mac/datawave.ingest.mapreduce.TableConfigHelperFactoryTest-testname
+ */
 public class TableConfigHelperFactoryTest {
     private static final Logger logger = Logger.getLogger(TableConfigHelperFactoryTest.class);
+    @Rule
+    public TestName name = new TestName();
     
     private Configuration conf;
-    private MockInstance instance;
+    private MiniAccumuloCluster mac;
     private Connector connector;
     private TableOperations tops;
     
     private static final String TEST_SHARD_TABLE_NAME = "testShard";
     
+    public void startCluster() throws Exception {
+        File macDir = new File(System.getProperty("user.dir") + "/target/mac/" + TableConfigHelperFactoryTest.class.getName() + "-" + name.getMethodName());
+        macDir.mkdirs();
+        mac = new MiniAccumuloCluster(new MiniAccumuloConfig(macDir, "pass"));
+        mac.start();
+    }
+    
     @Before
     public void setup() throws Exception {
+        startCluster();
```

Comment:
Is it possible to put this in a `@BeforeClass` and the shutdown in an `@AfterClass` method so we don't have to pay the startup/shutdown penalty on every test method? Looks like it's only 2 in this class, but the change would then let it run twice as fast.

Reply:
Yeah I just had to make each test delete and create the tables and make everything static.

Reply:
Tests takes about 5s now on my dev box.